M116 public

HinTak opened the pull request:
As the title says. I disabled about two tests which are too broken (segfaults) but the rest is 113 failed, 1974 passed, 34 skipped and 8 xfailed, 66 errors.

You may need to adjust the non-linux build instructions.

I only have one package which depends on skip-python and this works okay as a drop-in for m87.

I don't know how far/fast I can do the rest, but this is useful as is.

I have a detached mirror called skia-m1xx-python to add a few more interesting stuff for my own use; file issues and track progress, etc.

HinTak commented:
@kyamagu the 2nd push (re-enabling gn build and the admin task) seems to get CI to start building gn! Let's hope it continues to building skia. The first CI run failed with some syntax error - what was it?

HinTak commented:
3rd push does it - it is half way building skia! The build host is quite old software wise , still using python 2. Hence have to ask for python 3 explicitly in the build script.

HinTak commented:
CI failed in the middle of building skia due to the host's freetype being too old to support skia m116. Switching to build bundled freetype.

HinTak commented:
Aarch64 linux build was finally successful, build time about 4 hours.

HinTak commented:
Ubuntu auto64 fails at starting xvfb. Some problem with the build system/config itself.

HinTak commented:
Windows fails to compile GL related stuff. I think I know why.
```
[Code](https://github.com/kyamagu/skia-python)
Issues
37
Pull requests
2
[Actions](https://github.com/kyamagu/skia-python/actions)
[Projects](https://github.com/kyamagu/skia-python/projects)
[Security](https://github.com/kyamagu/skia-python/security)
[Insights](https://github.com/kyamagu/skia-python/pulse)
[Back to pull request #196](https://github.com/kyamagu/skia-python/pull/196)
M116 public #80
Build wheels on windows-2022 (auto64) for cp3{7,8,9,10,11}
failed 3 hours ago in 18m 12s
5s
24s
0s
0s
17m 41s
    
    × python setup.py bdist_wheel did not run successfully.
    │ exit code: 1
    ╰─> [39 lines of output]
        C:\Users\runneradmin\AppData\Local\Temp\cibw-run-rh1y1swv\cp37-win_amd64\build\venv\lib\site-packages\setuptools\__init__.py:85: _DeprecatedInstaller: setuptools.installer and fetch_build_eggs are deprecated. Requirements should be satisfied by a PEP 517 installer. If you are using pip, you can try `pip install --use-pep517`.
          dist.fetch_build_eggs(dist.setup_requires)
        running bdist_wheel
        running build
        running build_ext
        No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils
        building 'skia' extension
        creating build
        creating build\temp.win-amd64-cpython-37
        creating build\temp.win-amd64-cpython-37\Release
        creating build\temp.win-amd64-cpython-37\Release\src
        creating build\temp.win-amd64-cpython-37\Release\src\skia
        C:\Users\runneradmin\AppData\Local\Temp\cibw-run-rh1y1swv\cp37-win_amd64\build\venv\lib\site-packages\setuptools\_distutils\_msvccompiler.py:535: UserWarning: Fallback spawn triggered. Please update distutils monkeypatch.
          warnings.warn("Fallback spawn triggered. Please update distutils monkeypatch.")
        C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\bin\HostX86\x64\cl.exe /c /nologo /O2 /W3 /GL /DNDEBUG /MD -IC:\Users\runneradmin\AppData\Local\Temp\cibw-run-rh1y1swv\cp37-win_amd64\build\venv\lib\site-packages\pybind11\include -IC:\Users\runneradmin\AppData\Local\Temp\cibw-run-rh1y1swv\cp37-win_amd64\build\venv\lib\site-packages\pybind11\include -Iskia -Iskia\out\Release\gen -IC:\Users\runneradmin\AppData\Local\Temp\cibw-run-rh1y1swv\cp37-win_amd64\build\venv\include -IC:\Users\runneradmin\AppData\Local\pypa\cibuildwheel\Cache\nuget-cpython\python.3.7.9\tools\include -IC:\Users\runneradmin\AppData\Local\pypa\cibuildwheel\Cache\nuget-cpython\python.3.7.9\tools\Include -IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\include -IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\ATLMFC\include -IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\VS\include -IC:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\ucrt -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\shared -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\winrt -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\cppwinrt -IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um /EHsc /Tpsrc\skia\Bitmap.cpp /Fobuild\temp.win-amd64-cpython-37\Release\src\skia\Bitmap.obj /std:c++17 /DVERSION_INFO=116.0 /DSK_GL /Zc:inline /wd5030 /wd4244 /wd4267 /wd4800 /wd4180 /MD
        C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\bin\HostX86\x64\cl.exe /c /nologo /O2 /W3 /GL /DNDEBUG /MD -IC:\Users\runneradmin\AppData\Local\Temp\cibw-run-rh1y1swv\cp37-win_amd64\build\venv\lib\site-packages\pybind11\include -IC:\Users\runneradmin\AppData\Local\Temp\cibw-run-rh1y1swv\cp37-win_amd64\build\venv\lib\site-packages\pybind11\include -Iskia -Iskia\out\Release\gen -IC:\Users\runneradmin\AppData\Local\Temp\cibw-run-rh1y1swv\cp37-win_amd64\build\venv\include -IC:\Users\runneradmin\AppData\Local\pypa\cibuildwheel\Cache\nuget-cpython\python.3.7.9\tools\include -IC:\Users\runneradmin\AppData\Local\pypa\cibuildwheel\Cache\nuget-cpython\python.3.7.9\tools\Include -IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\include -IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\ATLMFC\include -IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\VS\include -IC:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\ucrt -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\shared -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\winrt -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\cppwinrt -IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um /EHsc /Tpsrc\skia\BlendMode.cpp /Fobuild\temp.win-amd64-cpython-37\Release\src\skia\BlendMode.obj /std:c++17 /DVERSION_INFO=116.0 /DSK_GL /Zc:inline /wd5030 /wd4244 /wd4267 /wd4800 /wd4180 /MD
        C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\bin\HostX86\x64\cl.exe /c /nologo /O2 /W3 /GL /DNDEBUG /MD -IC:\Users\runneradmin\AppData\Local\Temp\cibw-run-rh1y1swv\cp37-win_amd64\build\venv\lib\site-packages\pybind11\include -IC:\Users\runneradmin\AppData\Local\Temp\cibw-run-rh1y1swv\cp37-win_amd64\build\venv\lib\site-packages\pybind11\include -Iskia -Iskia\out\Release\gen -IC:\Users\runneradmin\AppData\Local\Temp\cibw-run-rh1y1swv\cp37-win_amd64\build\venv\include -IC:\Users\runneradmin\AppData\Local\pypa\cibuildwheel\Cache\nuget-cpython\python.3.7.9\tools\include -IC:\Users\runneradmin\AppData\Local\pypa\cibuildwheel\Cache\nuget-cpython\python.3.7.9\tools\Include -IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\include -IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\ATLMFC\include -IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\VS\include -IC:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\ucrt -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\shared -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\winrt -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\cppwinrt -IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um /EHsc /Tpsrc\skia\Canvas.cpp /Fobuild\temp.win-amd64-cpython-37\Release\src\skia\Canvas.obj /std:c++17 /DVERSION_INFO=116.0 /DSK_GL /Zc:inline /wd5030 /wd4244 /wd4267 /wd4800 /wd4180 /MD
        C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\bin\HostX86\x64\cl.exe /c /nologo /O2 /W3 /GL /DNDEBUG /MD -IC:\Users\runneradmin\AppData\Local\Temp\cibw-run-rh1y1swv\cp37-win_amd64\build\venv\lib\site-packages\pybind11\include -IC:\Users\runneradmin\AppData\Local\Temp\cibw-run-rh1y1swv\cp37-win_amd64\build\venv\lib\site-packages\pybind11\include -Iskia -Iskia\out\Release\gen -IC:\Users\runneradmin\AppData\Local\Temp\cibw-run-rh1y1swv\cp37-win_amd64\build\venv\include -IC:\Users\runneradmin\AppData\Local\pypa\cibuildwheel\Cache\nuget-cpython\python.3.7.9\tools\include -IC:\Users\runneradmin\AppData\Local\pypa\cibuildwheel\Cache\nuget-cpython\python.3.7.9\tools\Include -IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\include -IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\ATLMFC\include -IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\VS\include -IC:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\ucrt -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\shared -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\winrt -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\cppwinrt -IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um /EHsc /Tpsrc\skia\Codec.cpp /Fobuild\temp.win-amd64-cpython-37\Release\src\skia\Codec.obj /std:c++17 /DVERSION_INFO=116.0 /DSK_GL /Zc:inline /wd5030 /wd4244 /wd4267 /wd4800 /wd4180 /MD
        C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\bin\HostX86\x64\cl.exe /c /nologo /O2 /W3 /GL /DNDEBUG /MD -IC:\Users\runneradmin\AppData\Local\Temp\cibw-run-rh1y1swv\cp37-win_amd64\build\venv\lib\site-packages\pybind11\include -IC:\Users\runneradmin\AppData\Local\Temp\cibw-run-rh1y1swv\cp37-win_amd64\build\venv\lib\site-packages\pybind11\include -Iskia -Iskia\out\Release\gen -IC:\Users\runneradmin\AppData\Local\Temp\cibw-run-rh1y1swv\cp37-win_amd64\build\venv\include -IC:\Users\runneradmin\AppData\Local\pypa\cibuildwheel\Cache\nuget-cpython\python.3.7.9\tools\include -IC:\Users\runneradmin\AppData\Local\pypa\cibuildwheel\Cache\nuget-cpython\python.3.7.9\tools\Include -IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\include -IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\ATLMFC\include -IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\VS\include -IC:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\ucrt -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\shared -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\winrt -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\cppwinrt -IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um /EHsc /Tpsrc\skia\Color.cpp /Fobuild\temp.win-amd64-cpython-37\Release\src\skia\Color.obj /std:c++17 /DVERSION_INFO=116.0 /DSK_GL /Zc:inline /wd5030 /wd4244 /wd4267 /wd4800 /wd4180 /MD
        C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\bin\HostX86\x64\cl.exe /c /nologo /O2 /W3 /GL /DNDEBUG /MD -IC:\Users\runneradmin\AppData\Local\Temp\cibw-run-rh1y1swv\cp37-win_amd64\build\venv\lib\site-packages\pybind11\include -IC:\Users\runneradmin\AppData\Local\Temp\cibw-run-rh1y1swv\cp37-win_amd64\build\venv\lib\site-packages\pybind11\include -Iskia -Iskia\out\Release\gen -IC:\Users\runneradmin\AppData\Local\Temp\cibw-run-rh1y1swv\cp37-win_amd64\build\venv\include -IC:\Users\runneradmin\AppData\Local\pypa\cibuildwheel\Cache\nuget-cpython\python.3.7.9\tools\include -IC:\Users\runneradmin\AppData\Local\pypa\cibuildwheel\Cache\nuget-cpython\python.3.7.9\tools\Include -IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\include -IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\ATLMFC\include -IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\VS\include -IC:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\ucrt -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\shared -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\winrt -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\cppwinrt -IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um /EHsc /Tpsrc\skia\ColorFilter.cpp /Fobuild\temp.win-amd64-cpython-37\Release\src\skia\ColorFilter.obj /std:c++17 /DVERSION_INFO=116.0 /DSK_GL /Zc:inline /wd5030 /wd4244 /wd4267 /wd4800 /wd4180 /MD
        C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\bin\HostX86\x64\cl.exe /c /nologo /O2 /W3 /GL /DNDEBUG /MD -IC:\Users\runneradmin\AppData\Local\Temp\cibw-run-rh1y1swv\cp37-win_amd64\build\venv\lib\site-packages\pybind11\include -IC:\Users\runneradmin\AppData\Local\Temp\cibw-run-rh1y1swv\cp37-win_amd64\build\venv\lib\site-packages\pybind11\include -Iskia -Iskia\out\Release\gen -IC:\Users\runneradmin\AppData\Local\Temp\cibw-run-rh1y1swv\cp37-win_amd64\build\venv\include -IC:\Users\runneradmin\AppData\Local\pypa\cibuildwheel\Cache\nuget-cpython\python.3.7.9\tools\include -IC:\Users\runneradmin\AppData\Local\pypa\cibuildwheel\Cache\nuget-cpython\python.3.7.9\tools\Include -IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\include -IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\ATLMFC\include -IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\VS\include -IC:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\ucrt -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\shared -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\winrt -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\cppwinrt -IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um /EHsc /Tpsrc\skia\ColorSpace.cpp /Fobuild\temp.win-amd64-cpython-37\Release\src\skia\ColorSpace.obj /std:c++17 /DVERSION_INFO=116.0 /DSK_GL /Zc:inline /wd5030 /wd4244 /wd4267 /wd4800 /wd4180 /MD
        C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\bin\HostX86\x64\cl.exe /c /nologo /O2 /W3 /GL /DNDEBUG /MD -IC:\Users\runneradmin\AppData\Local\Temp\cibw-run-rh1y1swv\cp37-win_amd64\build\venv\lib\site-packages\pybind11\include -IC:\Users\runneradmin\AppData\Local\Temp\cibw-run-rh1y1swv\cp37-win_amd64\build\venv\lib\site-packages\pybind11\include -Iskia -Iskia\out\Release\gen -IC:\Users\runneradmin\AppData\Local\Temp\cibw-run-rh1y1swv\cp37-win_amd64\build\venv\include -IC:\Users\runneradmin\AppData\Local\pypa\cibuildwheel\Cache\nuget-cpython\python.3.7.9\tools\include -IC:\Users\runneradmin\AppData\Local\pypa\cibuildwheel\Cache\nuget-cpython\python.3.7.9\tools\Include -IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\include -IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\ATLMFC\include -IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\VS\include -IC:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\ucrt -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\shared -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\winrt -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\cppwinrt -IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um /EHsc /Tpsrc\skia\Data.cpp /Fobuild\temp.win-amd64-cpython-37\Release\src\skia\Data.obj /std:c++17 /DVERSION_INFO=116.0 /DSK_GL /Zc:inline /wd5030 /wd4244 /wd4267 /wd4800 /wd4180 /MD
        C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\bin\HostX86\x64\cl.exe /c /nologo /O2 /W3 /GL /DNDEBUG /MD -IC:\Users\runneradmin\AppData\Local\Temp\cibw-run-rh1y1swv\cp37-win_amd64\build\venv\lib\site-packages\pybind11\include -IC:\Users\runneradmin\AppData\Local\Temp\cibw-run-rh1y1swv\cp37-win_amd64\build\venv\lib\site-packages\pybind11\include -Iskia -Iskia\out\Release\gen -IC:\Users\runneradmin\AppData\Local\Temp\cibw-run-rh1y1swv\cp37-win_amd64\build\venv\include -IC:\Users\runneradmin\AppData\Local\pypa\cibuildwheel\Cache\nuget-cpython\python.3.7.9\tools\include -IC:\Users\runneradmin\AppData\Local\pypa\cibuildwheel\Cache\nuget-cpython\python.3.7.9\tools\Include -IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\include -IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\ATLMFC\include -IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\VS\include -IC:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\ucrt -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\shared -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\winrt -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\cppwinrt -IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um /EHsc /Tpsrc\skia\Document.cpp /Fobuild\temp.win-amd64-cpython-37\Release\src\skia\Document.obj /std:c++17 /DVERSION_INFO=116.0 /DSK_GL /Zc:inline /wd5030 /wd4244 /wd4267 /wd4800 /wd4180 /MD
        C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\bin\HostX86\x64\cl.exe /c /nologo /O2 /W3 /GL /DNDEBUG /MD -IC:\Users\runneradmin\AppData\Local\Temp\cibw-run-rh1y1swv\cp37-win_amd64\build\venv\lib\site-packages\pybind11\include -IC:\Users\runneradmin\AppData\Local\Temp\cibw-run-rh1y1swv\cp37-win_amd64\build\venv\lib\site-packages\pybind11\include -Iskia -Iskia\out\Release\gen -IC:\Users\runneradmin\AppData\Local\Temp\cibw-run-rh1y1swv\cp37-win_amd64\build\venv\include -IC:\Users\runneradmin\AppData\Local\pypa\cibuildwheel\Cache\nuget-cpython\python.3.7.9\tools\include -IC:\Users\runneradmin\AppData\Local\pypa\cibuildwheel\Cache\nuget-cpython\python.3.7.9\tools\Include -IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\include -IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\ATLMFC\include -IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\VS\include -IC:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\ucrt -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\shared -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\winrt -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\cppwinrt -IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um /EHsc /Tpsrc\skia\Font.cpp /Fobuild\temp.win-amd64-cpython-37\Release\src\skia\Font.obj /std:c++17 /DVERSION_INFO=116.0 /DSK_GL /Zc:inline /wd5030 /wd4244 /wd4267 /wd4800 /wd4180 /MD
        C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\bin\HostX86\x64\cl.exe /c /nologo /O2 /W3 /GL /DNDEBUG /MD -IC:\Users\runneradmin\AppData\Local\Temp\cibw-run-rh1y1swv\cp37-win_amd64\build\venv\lib\site-packages\pybind11\include -IC:\Users\runneradmin\AppData\Local\Temp\cibw-run-rh1y1swv\cp37-win_amd64\build\venv\lib\site-packages\pybind11\include -Iskia -Iskia\out\Release\gen -IC:\Users\runneradmin\AppData\Local\Temp\cibw-run-rh1y1swv\cp37-win_amd64\build\venv\include -IC:\Users\runneradmin\AppData\Local\pypa\cibuildwheel\Cache\nuget-cpython\python.3.7.9\tools\include -IC:\Users\runneradmin\AppData\Local\pypa\cibuildwheel\Cache\nuget-cpython\python.3.7.9\tools\Include -IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\include -IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\ATLMFC\include -IC:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\VS\include -IC:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\ucrt -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\shared -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\winrt -IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\cppwinrt -IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um /EHsc /Tpsrc\skia\GrContext.cpp /Fobuild\temp.win-amd64-cpython-37\Release\src\skia\GrContext.obj /std:c++17 /DVERSION_INFO=116.0 /DSK_GL /Zc:inline /wd5030 /wd4244 /wd4267 /wd4800 /wd4180 /MD
        GrContext.cpp
        src\skia\GrContext.cpp(267): error C2039: 'asGLFormat': is not a member of 'GrBackendFormat'
        skia\include/gpu/GrBackendSurface.h(63): note: see declaration of 'GrBackendFormat'
        src\skia\GrContext.cpp(245): error C2065: 'asGLFormat': undeclared identifier
        src\skia\GrContext.cpp(297): error C2039: 'getGLTextureInfo': is not a member of 'GrBackendTexture'
        skia\include/gpu/GrBackendSurface.h(247): note: see declaration of 'GrBackendTexture'
        src\skia\GrContext.cpp(279): error C2065: 'getGLTextureInfo': undeclared identifier
        src\skia\GrContext.cpp(300): error C2039: 'glTextureParametersModified': is not a member of 'GrBackendTexture'
        skia\include/gpu/GrBackendSurface.h(247): note: see declaration of 'GrBackendTexture'
        src\skia\GrContext.cpp(279): error C2065: 'glTextureParametersModified': undeclared identifier
        src\skia\GrContext.cpp(380): error C2039: 'getGLFramebufferInfo': is not a member of 'GrBackendRenderTarget'
        skia\include/gpu/GrBackendSurface.h(453): note: see declaration of 'GrBackendRenderTarget'
        src\skia\GrContext.cpp(358): error C2065: 'getGLFramebufferInfo': undeclared identifier
```

HinTak commented:
Mac os x is a python2 VS python3 problem:

```
bash scripts/build_macOS.sh
    File "tools/git-sync-deps", line 245
      nonlocal anything_failed
                             ^
  SyntaxError: invalid syntax
```

HinTak commented:
@kyamagu windows also finishes building now, just fixing mac build now. The rest of the failure is just pytest does not pass yet. My m116-private is reaching  2000+ passes now, the rest is mostly API changes - so I am tempted to disable all the tests that don't pass for the time being and come back to them later. However, there are two segfaults which aren't API changes, and they must be fixed before m116 goes out, i think. One of them concerns the optional arguments you did in #51 - if I partially revert that change, I get rid of the segfault and also passes a bunch of tests! Can you have a look? The test is currently disabled and marked with segfault in the commit message, so quite easy to find. The other segfault is in font getKerningAdustment. I don't understand why they segfault - the skia API have not changed; so it might be a c++14 vs c++17 thing.

HinTak commented:
Considering this is a distinctly different version, people can roll back on pip etc, so I think it can go out and let people roll back if they don't like it... @kyamagu what do you think? The segfault still needs fixing though...

HinTak commented:
Okay, I downloaded the last finished CI log for the latest one, since the details looks interesting. Mac is failing build due to c++14 to c++17, should be fixed in the current running CI. I found something interesting:

Ubuntu linux's are doing
111 failed, 1976 passed, 33 skipped, 8 xfailed, 66 errors

but windows is
87 failed, 1772 passed, 269 skipped, 2 xfailed, 1 warning, 64 errors

So I guess I could have a look and skip those 111 + 66 too (that likely includes the windows 87 + 1 + 64) to force it to pass.

I am inclined to release this, but let people continue with continue/roll back to 87.x, if pypi allows multiple versions and roll-backs.

kyamagu commented:
@HinTak Thanks for your hard work!

> I am inclined to release this, but let people continue with continue/roll back to 87.x, if pypi allows multiple versions and roll-backs.

As there are a bunch of test failures yet, my suggestion is to go with a beta release; e.g., `116.b1`. That way, users can install a pre-release version only if they specify `--pre` option in `pip` while development continues.

- https://setuptools.pypa.io/en/latest/userguide/distribution.html

For a stable release, I would say at least test failures should be all resolved (some can be a skip for reasons) and there should be reasonable documentation that describes the breaking API changes.

HinTak commented:
> @HinTak Thanks for your hard work!
> 
> > I am inclined to release this, but let people continue with continue/roll back to 87.x, if pypi allows multiple versions and roll-backs.
> 
> As there are a bunch of test failures yet, my suggestion is to go with a beta release; e.g., `116.b1`. That way, users can install a pre-release version only if they specify `--pre` option in `pip` while development continues.
> 
> * https://setuptools.pypa.io/en/latest/userguide/distribution.html
> 
> For a stable release, I would say at least test failures should be all resolved (some can be a skip for reasons) and there should be reasonable documentation that describes the breaking API changes.

That's a even better idea than mine - to be honest, I just like to make it possible for others to try, while some of the breakages might take some time to fix. It might take a very long time to fix :-(.

The documentation part is easier - I keep a file README.m116 about new stuff I added anyway, so I can just diff m87 against m116 and write a bit more about it. Like all the emulate old API stuff (that probably should go away) and ask people not to depend on those.

I just tagged and pushed rc2 - 96 failed, 2030 passed, 34 skipped, 8 xfailed, and 27 errors. I think most of the remaining failures and errors are API changes and disappearing APIs. Anyway, what worry me most are the two tests that segfaults which I disabled.

HinTak commented:
So mac os build also works, but Windows build got broken by one recent change since last success. Trying to unbreakable it with a new commit. Mac shows:

73 failed, 1823 passed, 269 skipped, 2 xfailed, 27 errors

I only have a small number of further changes now. So @kyamagu : how to do a beta release?

kyamagu commented:
@HinTak The release process is the following.

1. Update the version in `setup.py`
2. Merge the commits into the main branch
3. [Create a release](https://github.com/kyamagu/skia-python/releases)

Release creation will trigger a CI workflow without skipping [the `publish` job](https://github.com/kyamagu/skia-python/blob/main/.github/workflows/ci.yml) that uploads wheels to PyPI. This means, to publish a wheel, the CI process must succeed without any errors. Perhaps we have to skip many tests explicitly.

HinTak commented:
@kyamagu Thanks. I already changed the version to "116.0b1" two commits ago (it was "116.0" at the beginning). I was going to ask about your preference - cosmetically I could disable all the remaining failed tests, it is 70 failed and 8 errors - so it looks pretty on this pull, or I could do nothing, and "not polute the commit history". You answered that question for me. In fact, the successive blocks of commit I did was branched off before the last "bulk fixes comment-out" of the previous, so there is a lineage of history which contains no "work arounds for things that don't (yet) work" . I gave up towards end of the rc3 tag.

But how do people get a "87.6"? It might be months or years before this is mature enough to replace 87.5 :-(.

kyamagu commented:
@HinTak 

> But how do people get a "87.6"?

Are you thinking of supporting multiple skia versions? It is possible to change the release workflow to support that, but I would say considering the required development effort we can discontinue supporting 87.x versions and completely move to 116.x.

HinTak commented:
@kyamagu I am just thinking that it will take months if not years to address all the m116 'REVISIT' skips. Meanwhile, some might like 87.5 plus some small changes / bug fixes. That said, I only have one package which depends on skia-python in my hard drive. It runs just fine without noticeable issue; and a couple of scripts I wrote (which was how/why I filed those issues of mine), which also works, and they will potentially now be better and moving forwards, as I slipped fix to #194 and #192 to this pull too.

HinTak commented:
I mentioned I have a detached mirror. Here is the release note I wrote which tries to immitate your style.
https://github.com/HinTak/skia-m1xx-python/releases/tag/[v116.0b1](https://github.com/HinTak/skia-m1xx-python/releases/tag/v116.0b1) . Apparently this needs somebody else from me review/approving before it can be merged.

HinTak commented:
@kyamagu @0lru all checks passed. The final count is that I have disabled about 80 tests with "m116:REVISIT", another 30 with "may not need REVISIT" and two "segfault:REVISIT". Out of about 2200 tests. Considering about 20% of API changed between m87 and m116, I may have fixed up about 15% of those and brought it up to about 5% broken/removed/disabled. That sounds about right. Just need an approving review to release.

My detached mirror's CI is functional - it only kicked in finally when I made the 116.0b1 release (and failing at the publishing stage, quite understandably). That means I can try things out with a pull there on platform I don't have, before putting it here, for future updates...if there is any :-(.

kyamagu commented on README.m116.md:
```diff
@@ -0,0 +1,81 @@
+* [SkSVGDOM::renderNode() is not exposed in python](https://github.com/kyamagu/skia-python/issues/192)
+* [three-args contructor to SkMemoryStream not exposed.](https://github.com/kyamagu/skia-python/issues/194)
+
+The SVG mododule left experimental in `m88` upstream. It has received many improvements since.
```
module

HinTak replied:
Thanks. Fixed in private branch.

kyamagu commented on src/skia/ColorFilter.cpp:
```diff
@@ -130,14 +135,21 @@ py::class_<SkColorMatrix>(m, "ColorMatrix")
     ;
 
 py::class_<SkColorFilters>(m, "ColorFilters")
+/*
```
Revisit, right?

HinTak replied:
This particular one, has a revisit at tests/test_colorfilter.py at line 50. So it will be looked at, at some point (no promises about time scale...), when the test get looked at. In the m116-public-try branch on my repo, I have a few, which either get re-enbled (if the code can be modified/updated to work), or switch to xfail, and a n additional comment added above the "/*" saying or quoting upstream release notes.

In general, it is possible I have commented out code that don't compile anymore without an associated "revisit" test. At some point I'll diff against v87.5 and double-check. At least my comment-out style is consistent. So they can be grep with `grep '/\*`. I checked it that way in the "no workaround" lineage, with "git diff v87.5... | grep '/\*'" until it got too inconvenient to insist on a no-workaround lineage...

HinTak replied:
So in the end, I expect every one of "grep -1 "/\*" src/skia/*" to show a previous line about why it was gone. Don't know how long it will take though. :-(. I hope this is an answer you can accept.

kyamagu commented on src/skia/SamplingOptions.cpp:
```diff
@@ -0,0 +1,16 @@
+            "interporate between 2x2 sample points (bilinear interpolation)")
+        .export_values();
+
+    samplingoptions
```
Is there a test case for this?

HinTak replied:
No, but the tests have been updated with new 'skia.SamplingOptions()' arguments where they are needed. The enum values above is needed as a new argument to one of the surface/canvas(?) Routines. I don't want to add new code to access the internals or the non-default constructors of this class. (If nobody asks for it, a bare default no-args constructor would probably make most people happy)

HinTak replied:
I can add a quick "is_instance" test if you insist?

HinTak replied:
I have staged a basic isinstance test over at https://github.com/HinTak/skia-m1xx-python/pull/1 . Will cherry-pick when CI passes over there.

kyamagu reviewed (commented):
Looks great! I've left just a few comments.

kyamagu commented:
@HinTak Just to confirm one more, have you checked the documentation build? Might be good to 1) minimally update the `README.md`, and 2) include `README.m116.md` in a new section of the `docs/`, like release notes, changes, or migration guide.

HinTak commented:
> @HinTak Just to confirm one more, have you checked the documentation build? Might be good to 1) minimally update the `README.md`, and 2) include `README.m116.md` in a new section of the `docs/`, like release notes, changes, or migration guide.

The build docs part of CI passed. I actually haven't touched the docs directory at all. TL;DR, I think as far as migration goes, some needs a "skis.SamplingOption" inserted, a even smaller set of people will notice planarConfig becomes two. Don't know how many will notice things gone and missed them - hence the "attributeerror" mention.

HinTak commented:
I am staging some updates at https://github.com/HinTak/skia-m1xx-python/pull/1 . This allows me to cancel etc, without polluting your workflow too much. HEAD over there is identical to this pull, do pull-1 is update to this (via cherry-pick eventually, collapsing history and removing wrong turns, etc), except the 2nd commit, which enables CI over there, which will never be cherry-pick'ed into this.

HinTak commented:
@kyamagu bunch of small changes, typos, a few "revisit" switched to xfail when there is a upstream release note item, and a few "revisit" removed with the corresponding code re-enabled/changed to work. Bumped to 116.0b2 . I think this is it. I think the next one is probably 117.0b2 as m117 should be out any time now :-(.

kyamagu reviewed (approved):
LGTM

Perhaps we should sort out the documentation in the next release

HinTak commented:
Thanks. Btw, the aarch64 build is taking close to 5.5 hours in some runs, so it might need to be split into two at some point. (I read it being special was because of a 6 hour limit). It might go over 6 at some point. One idea is to split the gn build which takes 1/2 hour into its own task. There is also the fact that it is the first and take 5 hours - would be nice to know if something is trivially broken (typo/missing braces) earlier, so it would be nice to move it later, or even it being triggered conditionally only occasionally. Anyway, I'll think about that and may try some ideas in my detach mirror.

kyamagu commented:
@HinTak Agreed. Perhaps it is possible to split the CI jobs into the following steps (within `cibuildwheel` containers), or even into a different repository dedicated to Skia build.
1. Build Skia dependencies
2. Build Skia
3. Build wheels

HinTak commented:
I checked my CI logs, the worst was 5 hours 52 minutes. :-(. Splitting gn out will split out only about half an hour. But ninja is clever enough to be interrupted / by stages - so it is possible to do `ninja -C out/Release third_party/dng_sdk` etc before the default target. That makes `build_Linux.sh` quite complicated - build gn, build 3rd-party, build skia.

Having a build 3rd-party would really help: icu and harfbuzz are both quite large, and 3rd party adds about 700 compilations to a total of 1900 I think. So having a `ninja -C out/Release third_party/*` in the middle should really help to cut the compilation time of skia by 1/3.
